crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -27,6 +27,31 @@ pub struct AnalyzerStatusParams {
     pub text_document: Option<TextDocumentIdentifier>,
 }
 
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct CrateInfoResult {
+    pub name: Option<String>,
+    pub version: Option<String>,
+    pub path: String,
```

Comment:
was changing this a to `TextDocumentIdentifier` still not possible?

Reply:
I really don't think that this should be a TextDocumentIdentifier, because it represents the path to the crate's directory, not to a text document.